### PR TITLE
reimplentation of #416 also addressing #413, #370 #362. 

### PR DESCRIPTION
**Pull request by attachmentgenie**
This patch implements a parameter and template to be able to set autosign entries from cod.

also addressing #413, #370 #362.

### CODE REVIEW

**Comment by mmoll** [comment]
This does not implement rolling out a script, which (I think) is the background of #416, but is looking good for the #362 usecase, so I'd like to get this in.

As mentioned in #370, just rolling out that file unconditionally is not what we want as default, as the smart-proxy will normally add its entries to that file and if these get overwritten by puppet we do have a serious problem. I think the contents of that file should *only* be manged, if the array is not an empty one (or something similar).

Also, please add tests.

**Comment by attachmentgenie** [comment]
fair enough comments. this does not address the intention of #416 only your comments to parameterise the content of the file.

I have now added some code to only use the template when autosign_entries is not empty and added some tests to validate this behavior.

**Comment by mmoll** [comment]
:+1:, IMHO this closes #362 and #370

File: templates/server/autosign.conf.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Managed by Puppet
+<% scope.lookupvar("puppet::server::autosign_entries").each do |entry| -%>
+<%= entry %>
+<% end -%>
```
Comment (ekohl) [inline templates/server/autosign.conf.erb]: nit: we have line endings in our files

**Comment by ekohl** [comment]
Other than the small nit :+1:

**Comment by attachmentgenie** [comment]
go on then, third time lucky.

**Comment by ekohl** [comment]
:+1:

**Comment by mmoll** [comment]
bedankt @attachmentgenie!

**Pull request closed without merging**